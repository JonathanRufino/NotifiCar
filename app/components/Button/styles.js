import { StyleSheet } from 'react-native';

import { Colors } from '../../commom';

const styles = StyleSheet.create({
    container: {
        backgroundColor: Colors.BLUE,
        borderRadius: 5,
        width: '100%',
        height: 50,
        justifyContent: 'center',
        alignItems: 'center',
    },
    text: {
        color: Colors.WHITE,
        fontWeight: '100',
        fontSize: 20,
    }
});

export default styles;
