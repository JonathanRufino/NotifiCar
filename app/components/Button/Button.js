import React, { Component } from 'react';
import { TouchableOpacity, Text } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';

class Button extends Component {
    render() {
        return (
            <TouchableOpacity
                style={this.props.containerStyle}
                onPress={this.props.onPress}
                activeOpacity={this.props.activeOpacity}
            >
                <Text style={this.props.textStyle}>
                    { this.props.title.toUpperCase() }
                </Text>
            </TouchableOpacity>
        );
    }
}

Button.propTypes = {
    title: PropTypes.string.isRequired,
    onPress: PropTypes.func.isRequired,
    activeOpacity: PropTypes.number,
    containerStyle: PropTypes.any,
    textStyle: PropTypes.any,
};

Button.defaultProps = {
    title: 'CustomButton',
    onPress: () => {},
    activeOpacity: 0.8,
    containerStyle: styles.container,
    textStyle: styles.text,
};

export default Button;
