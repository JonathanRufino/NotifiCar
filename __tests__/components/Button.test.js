import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../../app/components/Button';

describe('<Button />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <Button
                title='Test Button'
                onPress={() => console.log('Button pressed')}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
