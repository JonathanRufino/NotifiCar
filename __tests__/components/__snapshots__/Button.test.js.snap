// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#Eb7575",
      "borderRadius": 5,
      "elevation": 5,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFF",
        "fontSize": 20,
        "fontWeight": "400",
      }
    }
  >
    TEST BUTTON
  </Text>
</View>
`;
